### PR TITLE
Initial solution to smart backspace

### LINKED ISSUE #71506: Smart Backspace feature
There is a good example of how PHPStorm handles it, https://blog.jetbrains.com/phpstorm/2014/09/smart-backspace-in-phpstorm-8/ VSCode really lacks feature like this as it requires to delete all the spaces / tabs before the cursor when you want to go back to previous line end using backspace.

### PR DESCRIPTION
**Pull request by adrianp97**
We have implemented an initial version  of smart backspace. The user needs to do cmd + backspace (Mac) or shift + delete (windows) in order to delete all trailing whitespace on a single line.

This PR fixes #71506 (https://github.com/microsoft/vscode/issues/71506)

### CODE REVIEW

**Comment by adrianp97** [comment]
We have added this gif to show you the functionality we implement on MacOS. We press the "cmd + Delete" button for each delete.

deletedemo

File: src/vs/editor/contrib/linesOperations/linesOperations.ts [hunk 1]
```diff
@@ -613,6 +613,43 @@ export class DeleteAllLeftAction extends AbstractDeleteAllToBoundaryAction {
...
 
 		rangesToDelete.sort(Range.compareRangesUsingStarts);
+
+		let endColumn = selections[0].endColumn - 1;
+		let startColumn = selections[0].startColumn - 1;
+		let startLineNumber = selections[0].startLineNumber;
+		let endLineNumber = selections[0].endLineNumber;
+		let linesContent = model.getLineContent(startLineNumber);
+
+		let whitespace_case = (endColumn === 0) || (/^\s*$/.test(linesContent[endColumn - 1]));
```
Comment (chrants) [inline src/vs/editor/contrib/linesOperations/linesOperations.ts]: It would be nice if the variables here were declared in `camelCase`

**Review by alexdima — CHANGES_REQUESTED** [review]
Thank you for the PR. I have tried this out, but unfortunately it does not work really well. It doesn't work for multiple cursors, and also it does not work in simple cases like the following:

TO_UPLOAD

**Comment by alexdima** [comment]
Please reopen if you address the comments I left.

**Pull request closed without merging**